src/runtime/ltypes/ltypes.py

```diff
@@ -1,6 +1,53 @@
+from inspect import getfullargspec, getcallargs
+
+# data-types
+
 i32 = []
 i64 = []
 f32 = []
 f64 = []
 c32 = []
 c64 = []
+
+# overloading support
+
+class OverloadedFunction:
+    """
+    A wrapper class for allowing overloading.
+    """
+    global_map = {}
+
+    def __init__(self, func):
+        self.func_name = func.__name__
+        f_list = self.global_map.get(func.__name__, [])
+        f_list.append((func, getfullargspec(func)))
+        self.global_map[func.__name__] = f_list
+
+    def __call__(self, *args, **kwargs):
+        func_map_list = self.global_map.get(self.func_name, False)
+        if not func_map_list:
+            raise Exception("Function not defined")
+        for item in func_map_list:
+            func, key = item
+            try:
+                # This might fail for the cases when arguments don't match
+                ann_dict = getcallargs(func, *args, **kwargs)
+            except TypeError:
+                continue
```

Comment:
If the arguments don't match, shouldn't we rather raise an exception that the arguments don't match?

Reply:
Suppose, the list contains two functions `foo(a)` and `foo(a, b)`, then, `foo(1, 2)` will raise an error if we use `getcallargs` on the first function.

Reply:
Right. We should determine which of the overloads contains two integer arguments in this case. And just call that.

Reply:
Yes, that's right!